File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mini = Integer.MAX_VALUE;
        int maxi= 0;
        for(int i=0;i<prices.length;i++){
            if(prices[i]<mini){
                mini =prices[i];
            }
            int profit= prices[i]-mini;
            if(profit>maxi){
                maxi =profit;
            }
        }
        return maxi;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let mini = Infinity;
    let maxProfit = 0;
    for(let p of prices){
        if(p<mini){
            mini =p;
        }
        let profit = p-mini;
        maxProfit = Math.max(maxProfit, profit);
    }
    return maxProfit;
};